File: app/src/main/java/me/imzack/app/ender/view/callback/SingleTypePlanListItemTouchCallback.java
package me.imzack.app.ender.view.callback;

import android.support.v7.widget.helper.ItemTouchHelper;

import me.imzack.app.ender.model.bean.Plan;

import java.util.List;

public class SingleTypePlanListItemTouchCallback extends BaseItemTouchCallback {

    private List<Plan> mSingleTypePlanList;

    public SingleTypePlanListItemTouchCallback(List<Plan> singleTypePlanList) {
        mSingleTypePlanList = singleTypePlanList;
    }

    @Override
    public int getMoveDirs(int position) {
        if (position == 0) return ItemTouchHelper.DOWN;
        if (position == mSingleTypePlanList.size() - 1) return ItemTouchHelper.UP;
        return ItemTouchHelper.UP | ItemTouchHelper.DOWN;
    }

    @Override
    public int getSwipeDirs(int position) {
        return ItemTouchHelper.START | ItemTouchHelper.END;
    }
}

File: app/src/main/java/net/zackzhang/app/end/view/callback/SingleTypePlanListItemTouchCallback.kt
package net.zackzhang.app.end.view.callback

import android.support.v7.widget.helper.ItemTouchHelper

import net.zackzhang.app.end.model.bean.Plan

class SingleTypePlanListItemTouchCallback(private val mSingleTypePlanList: List<Plan>) : BaseItemTouchCallback() {

    override fun getMoveDirs(position: Int) = when (position) {
        0 -> ItemTouchHelper.DOWN
        mSingleTypePlanList.size - 1 -> ItemTouchHelper.UP
        else -> ItemTouchHelper.UP or ItemTouchHelper.DOWN
    }

    override fun getSwipeDirs(position: Int) = ItemTouchHelper.START or ItemTouchHelper.END
}
